substrate/client/network/src/litep2p/shim/notification/peerset.rs

```diff
@@ -924,6 +930,47 @@ impl Peerset {
 		}
 	}
 
+	/// Connect to all reserved peers.
+	///
+	/// Under the following conditions:
+	/// 1. The peer must be present to the current set of peers.
+	/// 2. The peer must be disconnected.
+	/// 3. The peer must not be banned.
+	///
+	/// All reserved peers returned are transitioned to the `PeerState::Opening` state.
+	fn connect_reserved_peers(&mut self) -> Vec<PeerId> {
+		self.reserved_peers
+			.iter()
+			.filter_map(|peer| {
+				if self.peerstore_handle.is_banned(peer) {
+					log::trace!(
+						target: LOG_TARGET,
+						"{}: Cannot connect to banned reserved {peer:?}",
+						self.protocol,
+					);
+					return None;
+				}
+
+				let peer_state = self.peers.get(peer);
+				if peer_state != Some(&PeerState::Disconnected) {
+					log::trace!(
+						target: LOG_TARGET,
+						"{}: Cannot connect to expected disconnected reserved {peer:?} state: {peer_state:?}",
+						self.protocol,
+					);
```

Comment:
Not sure if we should log this at all, as we are iterating over all reserved peers, and plenty of them can be expected to be connected. I was also thinking about getting the set difference between the old reserved peer set and the new one, and only going through it, like it's done for disconnecting the peers, but not sure if would be optimal here.